utils/python/CIME/task_maker.py

```diff
@@ -41,6 +41,9 @@ def __init__(self, case, remove_dead_tasks=False):
         self._compute_values()
 
     def _compute_values(self):
+        """
+        JGF: This code really needs a refactor/rethink
+        """
```

Comment:
Yeah, I think that when you have a function named `_compute_values`, it's a bit of a red flag that it might have kind of a vaguely-defined purpose...